pkg/framework/kind/type.cue

```diff
@@ -0,0 +1,140 @@
+
+// properties shared between all kind varieties.
+_sharedKind: {
+	// name is the canonical, machine-friendly name of a Kind.
+	//
+	// name is often used in generated code intended to be parsed be parsed or used
+	// by machines, so it follows the relatively strict DNS label naming standard,
+	// as defined in RFC 1123:
+	//  - Contain at most 63 characters
+	//  - Contain only lowercase alphanumeric characters or '-'
+	//  - Start with an alphabetic character
+	//  - End with an alphanumeric character
+	name: =~"^([a-z][a-z0-9-]{0,61}[a-z0-9])$"
+
+	// TODO need some fields related to other names, human names - pluralization, capitalization
+	// it's all necessary for conversion to CRDs
+
+	// maturity indicates the how far a given kind declaration is in its initial
+	// journey towards what might otherwise be referred to as 1.0.
+	maturity: "committed" | "experimental" | "stable" | "mature"
+
+	// The kind system itself is not mature enough yet for any particula
+	// kind to be called "mature."
+	 // TODO remove this once system is ready https://github.com/orgs/grafana/projects/133/views/8
+	maturity: "committed" | "experimental" | "stable"
+
+	form: "structured" | "raw"
+}
+
+// Raw is a variety of Kind that specifies a type for a raw file,
+// like an svg or parquet file. Grafana mostly acts as asset storage for raw
+// types: the byte sequence is a black box to Grafana, and type is determined
+// through metadata such as file extension.
+#Raw: {
+	_sharedKind
+	form: "raw"
+	extensions?: [...string]
+
+	maturity: *"experimental" | "mature" // TODO unclear if we want maturity for raw kinds
+
+	// Set this if there's a sanitize function that should be associated with this
+	// raw type. A call to the named function will be generated.
+	// TODO decide where this gets generated - almost certainly must be outside pkg/kind tree
+	//
+	// The function must conform to (TODO decide on a type signature)
+	sanitizeFuncName?: string
+
+	// need
+	// - sanitize function
+	// - get summary
+}
+
+// Structured is a variety of Kind where a schema specifies
+// validity rules for the byte sequence. These represent all the conventional
+// types and functional resources in Grafana, such as dashboards and datasources.
+//
+// Structured types may be defined either by Grafana itself (core types), or by
+// plugins. Plugin-defined types have a slightly reduced set of capabilities,
+// due to the constraints imposed by them being run in separate processes, and
+// the risks arising from executing code from potentially untrusted third parties.
+#Structured: {
+	_sharedKind
+	form:     "structured"
+
+	lineage:  thema.#Lineage
+
+	// currentVersion is the syntactic version of the lineage's latest schema.
+	currentVersion: thema.#SyntacticVersion & (thema.#LatestVersion & { lin: lineage }).out
+}
+
+// TODO
+#CoreStructured: {
+	#Structured
+	maturity: *"committed" | "experimental" | "stable" | "mature"
+}
+
+// TODO
+CustomStructured: {
```

Comment:
Should this be a definition?

Reply:
i actually just removed this comment, heh

```
// NOTE not a definition b/c apps/plugins could be written against a later Grafana version
```

Reply:
i think i'll switch it back to a definition, but explicitly marked open (`...`)